src/modules/settings/components/Table.jsx

```diff
@@ -0,0 +1,242 @@
+);
+
+class CustomWhisper extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSelectMenu = this.handleSelectMenu.bind(this);
+  }
+
+  handleSelectMenu(eventKey) {
+    const {onChange, rowData, dataKey} = this.props;
+    onChange(rowData.id, dataKey, eventKey);
+    this.trigger.hide();
+  }
+
+  render() {
+    const {children, options} = this.props;
+    return (
+      <Whisper
+        placement="autoVerticalStart"
+        trigger="click"
+        triggerRef={(ref) => {
+          this.trigger = ref;
+        }}
+        speaker={<MenuPopover options={options} onSelect={this.handleSelectMenu} />}>
+        {children}
+      </Whisper>
+    );
+  }
+}
+
+function useOutsideAlerter(ref, onClick, rowData) {
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        onClick('edit', rowData.id);
+        document.removeEventListener('mousedown', handleClickOutside);
+      }
+    }
+    document.addEventListener('mousedown', handleClickOutside);
+  }, [ref]);
+}
+
+const EditCell = ({rowData, dataKey, onChange, onMouseOver, onMouseLeave, onClick, ...props}) => {
+  switch (rowData.status) {
+    case 'HOVERING':
+      return (
+        <Cell
+          {...props}
+          className="table-content-editing"
+          onMouseLeave={() => onMouseLeave && onMouseLeave('hovering', rowData.id)}>
+          <input
+            className="bttv-rs-input"
+            style={{top: 7, height: 30, opacity: 0.5}}
+            defaultValue={rowData[dataKey]}
+            onClick={() => onClick('edit', rowData.id)}
+          />
+        </Cell>
+      );
+
+    case 'EDIT': {
+      const wrapperRef = useRef(null);
+      useOutsideAlerter(wrapperRef, onClick, rowData);
+      return (
+        <Cell {...props} className="table-content-editing">
+          <input
+            ref={wrapperRef}
+            className="bttv-rs-input"
+            style={{top: 7, height: 30}}
+            defaultValue={rowData[dataKey]}
+            onChange={(event) => onChange(rowData.id, dataKey, event.target.value)}
+          />
+        </Cell>
+      );
+    }
+
+    default:
+      return (
+        <Cell
+          {...props}
+          onMouseOver={() => onMouseOver && onMouseOver('hovering', rowData.id)}
+          style={{paddingTop: 13, paddingLeft: 10}}>
+          <span className="table-content-edit-span">{rowData[dataKey]}</span>
+        </Cell>
+      );
+  }
+};
+
+const ActionCell = ({rowData, dataKey, onClick, ...props}) => (
+  <Cell {...props} style={{padding: '6px 0'}}>
+    <Button style={{float: 'right', marginRight: 10}} appearance="link" onClick={() => onClick('delete', rowData.id)}>
+      <Icon icon={minus} />
+    </Button>
+  </Cell>
+);
+
+function EditTable({options, setData, data, ...props}) {
+  const modal = useContext(ModalContext);
+
+  if (!modal.enter) return <Placeholder.Graph active height={data.length * 46} />;
```

Comment:
nit: unless it's a single `return` or `continue` or `break` you should really use brackets.